nntrainer/src/layers.cpp

```diff
@@ -129,6 +129,24 @@ int Layer::setOptimizer(Optimizer opt) {
   return this->opt.initialize(height, width, true);
 }
 
+int Layer::checkValidation() {
+  int status = ML_ERROR_NONE;
+  if (!type || type == LAYER_UNKNOWN) {
+    ml_loge("Error: Layer type is unknown");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  if (!activation_type || activation_type == ACT_UNKNOWN) {
+    ml_loge("Error: Have to set activation for this layer");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+  if (!batch || !width || !height) {
```

Comment:
I think it is safer to validate `under 0` even though it is already checked at init phase.
```suggestion
  if (batch <= 0 || width <= 0 || height <= 0) {
```

Reply:
It's unsigned int type. I will change ```batch == 0```